src/latestCode.ts

```diff
@@ -10,20 +11,72 @@ export const latestCode = async (): Promise<Location> =>
       Bucket: DEPLOYMENT_BUCKET,
       Prefix: `serverless/webhook-handler/${ENV}/`,
     }
-    const err = new Error(`Code not in ${ps.Bucket}/${ps.Prefix}`)
-    const contents = (await s3.listObjectsV2(ps).promise()).Contents
-    if (!contents || !contents.length) throw err
+    const contents = await listObjects(ps)
 
-    const filtered = contents
-      .filter((c) => c.Key && c.Key.includes("func.zip"))
+    const sortedList = filterS3Objects(contents)
       .map((c) => c.Key as string)
       .sort((a, b) => (a > b ? -1 : 1))
-    if (!filtered.length) throw err
 
-    const key = filtered[0]
-    return {
-      bucket: ps.Bucket,
-      key,
-      version: key.split("/")[3].split("-").slice(1).join("-"),
-    }
+    const key = sortedList[0]
+
+    return s3ToLocation(ps.Bucket, key)
   })
+
+export const codeExists = async (codeVersion: string): Promise<Location> => {
+  const listRequest = {
+    Bucket: DEPLOYMENT_BUCKET,
+    Prefix: `serverless/webhook-handler/${ENV}/*-${codeVersion}/`,
```

Comment:
We are using a wildcard `*` here. Between amazon Q and docs online I am getting mixed messages if the wildcard works here. It will be quickly noticed in testing if the wildcard does not work. and we can run a filter similar to `Key.includes("func.zip")` with the codeVersion we want

Reply:
I am good to try in devint. It would not break anything just code wont be updated. There is logging to quickly identify the issue